plugins/mock/devicepluginmock.cpp

```diff
@@ -196,295 +274,493 @@ void DevicePluginMock::startMonitoringAutoDevices()
     QList<DeviceDescriptor> deviceDescriptorList;
     deviceDescriptorList.append(mockDescriptor);
 
-    emit autoDevicesAppeared(mockDeviceAutoDeviceClassId, deviceDescriptorList);
+    emit autoDevicesAppeared(deviceDescriptorList);
 }
 
-Device::DeviceSetupStatus DevicePluginMock::confirmPairing(const PairingTransactionId &pairingTransactionId, const DeviceClassId &deviceClassId, const ParamList &params, const QString &secret)
+void DevicePluginMock::startPairing(DevicePairingInfo *info)
 {
-    Q_UNUSED(params)
-    Q_UNUSED(secret)
+    if (info->deviceClassId() == mockPushButtonDeviceClassId) {
+        qCDebug(dcMockDevice) << "Push button. Pressing the button in 3 seconds.";
+        info->finish(Device::DeviceErrorNoError, QT_TR_NOOP("Wait 3 second before you continue, the push button will be pressed automatically."));
+        m_pushbuttonPressed = false;
+        QTimer::singleShot(3000, this, SLOT(onPushButtonPressed()));
+        return;
+    }
+
+    if (info->deviceClassId() == mockDisplayPinDeviceClassId) {
+        qCDebug(dcMockDevice) << "Display pin!! The pin is 243681";
+        info->finish(Device::DeviceErrorNoError, QT_TR_NOOP("Please enter the secret which normaly will be displayed on the device. For the mockdevice the pin is 243681."));
+        return;
+    }
+
+    if (info->deviceClassId() == mockUserAndPassDeviceClassId) {
+        qCDebug(dcMockDevice) << "User and password. Login is \"user\" and \"password\".";
+        info->finish(Device::DeviceErrorNoError, QT_TR_NOOP("Please enter login credentials for the mock device (\"user\" and \"password\")."));
+        return;
+    }
+
+    if (info->deviceClassId() == mockOAuthSonosDeviceClassId) {
+        QString clientId = "b15cbf8c-a39c-47aa-bd93-635a96e9696c";
+        QString clientSecret = "c086ba71-e562-430b-a52f-867c6482fd11";
+
+        QUrl url("https://api.sonos.com/login/v3/oauth");
+        QUrlQuery queryParams;
+        queryParams.addQueryItem("client_id", clientId);
+        queryParams.addQueryItem("redirect_uri", "https://127.0.0.1:8888");
+        queryParams.addQueryItem("response_type", "code");
+        queryParams.addQueryItem("scope", "playback-control-all");
+        queryParams.addQueryItem("state", "ya-ya");
+        url.setQuery(queryParams);
 
+        qCDebug(dcMockDevice()) << "Sonos url:" << url;
+
+        info->setOAuthUrl(url);
+        info->finish(Device::DeviceErrorNoError);
+        return;
+    }
+
+    if (info->deviceClassId() == mockOAuthGoogleDeviceClassId) {
+        QString clientId= "937667874529-pr6s5ciu6sfnnqmt2sppvb6rokbkjjta.apps.googleusercontent.com";
+        QString clientSecret = "1ByBRmNqaK08VC54eEVcnGf1";
+
```

Comment:
are those created with the official nymea account?

Reply:
No, it's an extra account created just for this, named "nymea (testing)"